velox/connectors/hive/storage_adapters/s3fs/S3Util.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Implementation of S3 filesystem and file interface.
+// We provide a registration method for read and write files so the appropriate
+// type of file can be constructed based on a filename. See the
+// (register|generate)ReadFile and (register|generate)WriteFile functions.
+
+#include "velox/connectors/hive/storage_adapters/s3fs/S3Util.h"
+
+namespace facebook::velox {
+
+std::string getErrorStringFromS3Error(
+    const Aws::Client::AWSError<Aws::S3::S3Errors>& error) {
+  switch (error.GetErrorType()) {
```

Comment:
These are the most common errors. We can add more mappings later based on the need.

Reply:
It would be nice to add tests for all of these errors.

Reply:
Different backend S3 services (Minio, AWS S3) will behave differently. I will investigate MinIO and see if it generates all these errors.

Reply:
I was able to test `Network Connection` by shutting down the MinIO server in one test. MinIO gave a generic `resource not found` for missing bucket or key. Also `service unavailable` seems to only happen when the server is on high-load, which will be hard to achieve here.